### PR TITLE
Issue #776 Builders javadoc

### PR DESCRIPTION
**Pull request by ljacomet**
* Remove javadoc for `internal` packages
* Javadoc for builders
* Misc clean ups

This is a first part of the work needed for #776, more to come.

### CODE REVIEW

File: 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java [hunk 1]
```diff
@@ -111,13 +111,13 @@
 
       CacheConfiguration<K, V> cacheConfiguration = builder.build();
 
-      if (!useJsr107Expiry) {
+      if (hasConfiguredExpiry) {
```
Comment (lorban) [inline 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java]: Why changing the test from `(!useJsr107Expiry)` to `(hasConfiguredExpiry)` instead of `(!hasConfiguredExpiry)`? An overlook maybe?

File: 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java [hunk 2]
```diff
@@ -111,13 +111,13 @@
...
         expiryPolicy = new EhcacheExpiryWrapper<K, V>(cacheConfiguration.getExpiry());
       }
 
       return new ConfigHolder<K, V>(
           new CacheResources<K, V>(cacheName, loaderWriter, expiryPolicy, initCacheEventListeners(jsr107Configuration)),
-          new Eh107CompleteConfiguration<K, V>(jsr107Configuration, cacheConfiguration, !useJsr107Expiry, useEhcacheLoaderWriter),
+          new Eh107CompleteConfiguration<K, V>(jsr107Configuration, cacheConfiguration, hasConfiguredExpiry, useEhcacheLoaderWriter),
```
Comment (lorban) [inline 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java]: Same as above, or have you spotted a bug?

File: xml/src/test/java/org/ehcache/xml/FromTemplateCacheConfigurationBuilderDefaultTest.java [hunk 3]
```diff
@@ -38,7 +38,7 @@ public void setUp() throws Exception {
   }
 
   @Test
-  public void testDefaultExpiry() throws Exception {
-    assertThat(minimalTemplateBuilder.hasDefaultExpiry(), is(true));
+  public void testNoConfiguredExpiry() throws Exception {
+    assertThat(minimalTemplateBuilder.hasConfiguredExpiry(), is(false));
```
Comment (lorban) [inline xml/src/test/java/org/ehcache/xml/FromTemplateCacheConfigurationBuilderDefaultTest.java]: What's the story behind this test change? Is that related to the inverted condition above?

File: 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java (hunk 1 above)
Comment (ljacomet) [inline 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java]: Because a configured expiry means we should not use the JSR-107 provided one. This stems from changing method name (and contract) on the builder: `CacheConfigurationBuilder.hasConfiguredExpiry()` vs the former `CacheConfigurationBuilder.hasDefaultExpiry`
When I was writing the javadoc, expressing the contract with default name was really cumbersome and felt way more natural with changing the name and meaning of `true` and `false`.

File: 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java (hunk 2 above)
Comment (ljacomet) [inline 107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java]: Same as above indeed - so not a bug but a desired change

File: xml/src/test/java/org/ehcache/xml/FromTemplateCacheConfigurationBuilderDefaultTest.java (hunk 3 above)
Comment (ljacomet) [inline xml/src/test/java/org/ehcache/xml/FromTemplateCacheConfigurationBuilderDefaultTest.java]: Correct
